python/ray/data/dataset.py

```diff
@@ -1589,44 +1598,44 @@ def train_test_split(
                     "If `test_size` is a float, it must be bigger than 0 and smaller "
                     f"than 1. Got {test_size}."
                 )
-            return dataset.split_proportionately([1 - test_size])
+            return ds.split_proportionately([1 - test_size])
         else:
-            dataset_length = dataset.count()
-            if test_size <= 0 or test_size >= dataset_length:
+            ds_length = ds.count()
+            if test_size <= 0 or test_size >= ds_length:
                 raise ValueError(
                     "If `test_size` is an int, it must be bigger than 0 and smaller "
-                    f"than the size of the dataset ({dataset_length}). "
+                    f"than the size of the datastream ({ds_length}). "
                     f"Got {test_size}."
                 )
-            return dataset.split_at_indices([dataset_length - test_size])
+            return ds.split_at_indices([ds_length - test_size])
 
     @ConsumptionAPI(pattern="Args:")
-    def union(self, *other: List["Dataset[T]"]) -> "Dataset[T]":
-        """Combine this dataset with others of the same type.
+    def union(self, *other: List["Datastream[T]"]) -> "MaterializedDatastream[T]":
```

Comment:
Can we keep the return value as `Datastream` instead of `MaterializedDatastream`? We will soon implement the union as a streaming physical operator (https://github.com/ray-project/ray/issues/32598) during Ray 2.5.